provider/oauth2/forms.py

```diff
@@ -340,10 +340,15 @@ def clean(self):
 
 class ClientCredentialsGrantForm(ScopeMixin, OAuthForm):
     """ Validate a client credentials grant request. """
+    scope = forms.CharField(required=False)
 
-    def clean(self):
-        cleaned_data = super(ClientCredentialsGrantForm, self).clean()
-        # We do not fully support scopes for this grant type; however, a scope is required
-        # in order to create an access token. Default to read-only access.
-        cleaned_data['scope'] = constants.READ
-        return cleaned_data
+    def clean_scope(self):
+        # NOTE (CCB): This is a horrible hack, like much of our OAuth work. The scopes are declared in
+        # edx-oauth2-provider. (See edx_oauth2_provider/constants.py.) However, we need to provide a default scope
+        # that (a) gives the token basic read access and (b) allows access to the user info endpoint. This value
+        # represents the following scopes: openid (1), profile (2), email (4), permissions (32). At present, this is
+        # all scopes except course_staff and course_instructor. These scopes are normally associated with actual
+        # users, whereas the client credentials grant will primarily be used by service users.
+        #
+        # In the future, we should limit the allowable scopes either at a global or per-client level.
+        return 39
```

Comment:
Why not use the actual scopes instead of a magic number with a lengthy comment describing it?
    
    return constants.OPENID | constants.PROFILE | constants.EMAIL | constants.PERMISSION

Are we trying to avoid making edx-django-oauth2-provider dependent on edx-oauth2-provider?